src/Util/Lock/MemcacheLockDriver.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace Friendica\Util\Lock;
+
+use Friendica\Core\Cache;
+use dba;
+
+class MemcacheLockDriver implements ILockDriver
+{
+	/**
+	 *
+	 * @brief Sets a lock for a given name
+	 *
+	 * @param string $key The Name of the lock
+	 * @param integer $timeout Seconds until we give up
+	 *
+	 * @return boolean Was the lock successful?
+	 */
+	public function acquireLock($key, $timeout = 120)
+	{
+		$got_lock = false;
+		$start = time();
+
+		$cachekey = get_app()->get_hostname() . ";lock:" . $key;
+
+		do {
+			// We only lock to be sure that nothing happens at exactly the same time
+			dba::lock('locks');
```

Comment:
I don't understand the existence of this DB call for this cache lock driver.

Reply:
I moved it from the original code where it was already. Maybe @annando knows more. I don't understand it too :)